### ISSUE TITLE
predict error

### ISSUE THREAD

**Issue by whatbb** [soda-inria/tabicl#15]
When I run
```
pred = clf.predict(test_X)
```
Jupyter will report an error:
```
The kernel for tabicl.ipynb appears to have died. It will restart automatically.
```
But my test set only has 1000 data points, so there is no issue of exceeding the memory limit.

**Comment by JingangQu** [comment]
Hi,

Could you please provide a minimal code snippet to help reproduce the issue?

Many thanks !

**Comment by JingangQu** [comment]
Hi,

I was able to reproduce the excessive memory usage with the following code, even when setting `batch_size=1`:

```python
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from tabicl import TabICLClassifier

X, y = make_classification(n_samples=38500, n_features=1432, random_state=42)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1000, stratify=y, random_state=42)

clf = TabICLClassifier(checkpoint_version="tabicl-classifier-v1.1-0506.ckpt", batch_size=1)
clf.fit(X_train, y_train)
predictions = clf.predict(X_test)
print("Accuracy:", accuracy_score(y_test, predictions))
```

A potential reason is that TabICL does not perform internal auto-batching when running on CPU, which leads to high memory consumption for large data. As a current workaround, we recommend using a GPU if available.

I'm investigating the issue further and will keep you updated.

**Comment by whatbb** [comment]
Okay, thank you very much for your answer.
